feat(eslint): Add space-before-function-paren

benji6 opened the pull request:
space-before-function-paren added and set to "never" https://eslint.org/docs/rules/space-before-function-paren

callbacksLast option for sort-prop-types has been removed https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-prop-types.md

Commit:
feat(eslint): Add space-before-function-paren
BREAKING CHANGE: space-before-function-paren added and set to "never"

andy-polhill reviewed (approved):
This shouldn't affect us much, but good to nail it down.

Commit:
feat(eslint): Remove callbacksLast option for react/sort-prop-types rule
BREAKING CHANGE: Sorting order for prop types is still alphabetical but no longer places callbacks last
